include/ginkgo/core/distributed/row_gatherer.hpp

```diff
@@ -129,24 +128,22 @@ class RowGatherer final
      * @tparam GlobalIndexType  the global index type of the index map
      *
      * @param exec  the executor
-     * @param coll_comm  the collective communicator
+     * @param coll_comm_creator  the collective communicator creator_fn
```

Comment:
could you also add the note here about revisiting this function argument and the reason?
I feel a bit weird when passing the function to create a object, but I do not have an idea to resolve the variant copy issue.

Reply:
I've reverted this change. There are still no temporary copies of the index map, and the collective communicator object is used again, instead of the creator function.